src/SingleCellAnnotator.imjoy.html

```diff
@@ -434,27 +450,39 @@
           body: JSON.stringify(sendData)
         });
       const res= await response.json(); 
-      if(!res.success){
+      if(res.success){
+        this.current_image_data.annotation_status=true;
+      }
+      else{
         alert('Failed to upload the annotaiton, error: ' + res.error);
         throw new Error('Failed to upload the annotaiton.')
       };
     },
     async annotation_mode_changed(){
       if(this.annotation_mode){
+        this.current_image_data.mask_adjusted=false;
         await this.extendImage();
+        this.removeMaskAnnotation(); // remove the current mask if adjust mask
         this.startAnnotation(this.vector_source, this.freehand_mode)
       }
       else{
+        if(this.current_image_data.mask_adjusted==false){
+          api.alert("mask status: " + this.current_image_data.mask_adjusted);
+          this.annotation_mode = true;
+          api.alert("Don't forget to draw or save drawn mask!")
```

Comment:
I think we don't need to block the user in this case, we should allow switching off the annotation adjustment mode without drawing a mask, in such case the mask won't be modified.

Reply:
Let's keep it